src/app/basketProduct/basketProduct.service.ts

```diff
@@ -11,16 +11,21 @@ export class BasketProductService {
   products: BasketProduct[] = [];
   productsId: number[] = [];
 
-  constructor() { }
+  constructor(private authentication: Auth0Service) { }
 
   getAllProducts(): BasketProduct[] {
+    return JSON.parse(localStorage.getItem('products'));
+  }
+  getProducts(): BasketProduct[] {
     return this.products;
   }
-
   addProduct(product: BasketProduct): void {
+
     if (this.productsId.indexOf(product.product.id) === -1) {
       this.products.push(product);
       this.productsId.push(product.product.id);
+      localStorage.setItem('products', JSON.stringify(this.products));
+      localStorage.setItem('productsId', JSON.stringify(this.productsId));
```

Comment:
An element (`products`) and its identifier (`productsId`) are always better to handle using a [dictionary](https://www.dotnetperls.com/object-js), rather than 2 arrays.